osquery/config/packs.cpp

```diff
@@ -216,7 +216,8 @@ const std::string& Pack::getVersion() const {
 }
 
 bool Pack::shouldPackExecute() {
-  return (valid_ && checkDiscovery());
+  executing_ = (valid_ && checkDiscovery());
+  return executing_.load();
```

Comment:
Is the goal for "should" to "do some work" and otherwise the accessor "is" to just inspect the cached result? If so, this looks good.

nit, can we ditch the "load" method for atomics?